receiver/jaegerreceiver/trace_receiver.go

```diff
@@ -441,7 +442,7 @@ func (jr *jReceiver) HandleThriftHTTPBatch(w http.ResponseWriter, r *http.Reques
 
 	batch, hErr := jr.decodeThriftHTTPBody(r)
 	if hErr != nil {
-		http.Error(w, hErr.msg, hErr.statusCode)
+		http.Error(w, html.EscapeString(hErr.msg), hErr.statusCode)
```

Comment:
nit: (Morally) shouldn't this be also done below on the `consumeTraces` error too? I don't think it poses a security risk now but it's better to be safe than sorry

Reply:
We are doing this here because this error message might include user-provided input, namely, the content-type header which is returned embedded into the errors from lines 411 and 417 from `decodeThriftHTTPBody`. The one you mentioned doesn't seem to embed any user-provided input.

Reply:
> The one you mentioned doesn't seem to embed any user-provided input.

Sure, not saying it does, just that it's easier/safer to escape things everywhere rather than doing it on a per-case basis. Just being nitpicky, feel free to dismiss this :)

Reply:
There was a warning about this specific instance in an automated security checking tool, so I would like to fast-track it. But in general, I'd agree that everything that goes back to clients should be properly escaped. Perhaps we could have a broader discussion/guideline for this, as I suspect other components might be affected too.